src/js/util/object.js

```diff
@@ -0,0 +1,53 @@
+function isObject(something) {
+  return typeof something === 'object';
+}
+
+/**
+ *
+ * @param {Object} object
+ */
+function cloneDeepObject(object) {
+  const keys = Object.keys(object);
+  return Object.freeze(
+    keys.reduce((result, key) => {
+      const value = object[key];
+      if (Array.isArray(value)) {
+        return { ...result, [key]: [...value] };
+      }
+      if (typeof value === 'object') {
+        return { ...result, [key]: Object.freeze({ ...cloneDeepObject(value) }) };
+      }
+      return { ...result, [key]: value };
+    }, {})
```

Comment:
```suggestion
    keys.reduce((result, key) => {
      const value = object[key];
      const copiedValue = Array.isArray(value)      ? cloneDeepArray(value)  :
                          typeof value === 'object' ? cloneDeepObject(value) :  value;
                                          
      return { ...result, [key]: value };
    }, {});
```
이럴 때는 중첩 삼항연산자를 사용해도 깔끔할 것 같아요!
다만 prettier ignore가 필요하겠네요 ㅋㅋ

Reply:
제안해주신 부분을 중첩 삼항연산자를 적용하는 데는 조금 무리가 있어 보입니다.
되도록이면 일관성을 지키고 예외 케이스를 최소화하고자 합니다.

1. prettier ignore를 위한 예외 케이스를 작성하는 것도 개인적으로는 꺼림칙합니다.
2. airbnb의 코드 스타일 가이드를 지향하고 있어서 중첩 삼항연산자는 지양하고자 합니다.
> [15.6](https://github.com/airbnb/javascript#comparison--nested-ternaries) Ternaries should not be nested and generally be single line expressions. eslint: [no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary)

가독성 vs 명확성으로 팽팽하게 대립하는데 이 부분에 관해서는 개인적인 호불호가 갈리는 부분이어서 일관성을 가져가는 쪽으로 진행할게요! 새로운 관점을 제시해주셔서 감사합니다